channel_notifier.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/chanbackup"
+	"github.com/lightningnetwork/lnd/channelnotifier"
+)
+
+// addrSource is an interface that allow us to get the addresses for a target
+// node. We'll need this in order to be able to properly proxy the
+// notifications to create SCBs.
+type addrSource interface {
+	// AddrsForNode returns all known addresses for the target node public
+	// key.
+	AddrsForNode(nodePub *btcec.PublicKey) ([]net.Addr, error)
+}
+
+// channelNotifier is an implementation of the chanbackup.ChannelNotifier
+// interface using the existing channelnotifier.ChannelNotifier struct. This
+// implementation allows us to satisfy all the dependencies of the
+// chanbackup.SubSwapper struct.
+type channelNotifier struct {
+	// chanNotifier is the based channel notifier that we'll proxy requests
+	// from.
+	chanNotifier *channelnotifier.ChannelNotifier
+
+	// addrs is an implementation of the addrSource interface that allows
+	// us to get the latest set of addresses for a given node. We'll need
+	// this to be able to create an SCB for new channels.
+	addrs addrSource
+}
+
+// SubscribeChans requests a new channel subscription relative to the initial
+// set of known channels. We use the knownChans as a synchronization point to
+// ensure that the chanbackup.SubSwapper does not miss any channel open or
+// close events in the period between when it's created, and when it requests
+// the channel subscription.
+//
+// NOTE: This is part of the chanbackup.ChannelNotifier interface.
+func (c *channelNotifier) SubscribeChans(startingChans map[wire.OutPoint]struct{}) (
```

Comment:
`startingChans` now used?

Reply:
Only there to meet the interface, but it isn't being used atm as we send ourselves the fresh set of channels at start up now.